framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/util/PropertyLoader.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.cloud.openshift.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.NoSuchFileException;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PropertyLoader {
+
+    private static final Logger log = LoggerFactory.getLogger(PropertyLoader.class);
+
+    private PropertyLoader() {
+        // Util class
+    }
+
+    public static Properties loadProperties(String resourceAbsoluteFilename) {
+        Properties properties = loadPropertiesFromResource(resourceAbsoluteFilename);
+        return filterProperties(properties);
+    }
+
+    private static Properties loadPropertiesFromResource(String resourceAbsoluteFilename) {
+        Properties properties = new Properties();
+        try (InputStream is = PropertyLoader.class.getResourceAsStream(resourceAbsoluteFilename)) {
+            if (is == null) {
+                throw new NoSuchFileException(resourceAbsoluteFilename);
+            }
+            properties.load(is);
+            log.info("Loaded {} propert{} from {}",
+                     properties.size(),
+                     properties.size() == 1 ? "y" : "ies",
+                     resourceAbsoluteFilename
+            );
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to load properties from " + resourceAbsoluteFilename, e);
+        }
+        return properties;
+    }
+
+    /**
+     * Process properties and replace any variable with appropriate value. The value can be retrieved from another property in properties.
+     *
+     * @param properties Properties to be processed.
+     * @return Processed properties.
+     */
+    private static Properties filterProperties(Properties properties) {
+        Pattern replaceTagPattern = Pattern.compile(".*\\$\\{(.*)\\}.*");
```

Comment:
Perpahs compile just once and place the pattern in a static field?

Reply:
Good tip, thanks. Implemented